cloudinit/distros/freebsd.py

```diff
@@ -282,309 +185,16 @@ def create_user(self, name, **kwargs):
             keys = set(kwargs['ssh_authorized_keys']) or []
             ssh_util.setup_user_keys(keys, name, options=None)
 
-    @staticmethod
-    def get_ifconfig_list():
-        cmd = ['ifconfig', '-l']
-        (nics, err) = util.subp(cmd, rcs=[0, 1])
-        if len(err):
-            LOG.warning("Error running %s: %s", cmd, err)
-            return None
-        return nics
-
-    @staticmethod
-    def get_ifconfig_ifname_out(ifname):
-        cmd = ['ifconfig', ifname]
-        (if_result, err) = util.subp(cmd, rcs=[0, 1])
-        if len(err):
-            LOG.warning("Error running %s: %s", cmd, err)
-            return None
-        return if_result
-
-    @staticmethod
-    def get_ifconfig_ether():
-        cmd = ['ifconfig', '-l', 'ether']
-        (nics, err) = util.subp(cmd, rcs=[0, 1])
-        if len(err):
-            LOG.warning("Error running %s: %s", cmd, err)
-            return None
-        return nics
-
-    @staticmethod
-    def get_interface_mac(ifname):
-        if_result = Distro.get_ifconfig_ifname_out(ifname)
-        for item in if_result.splitlines():
-            if item.find('ether ') != -1:
-                mac = str(item.split()[1])
-                if mac:
-                    return mac
-
-    @staticmethod
-    def get_devicelist():
```

Comment:
other parts of the code are still using this call, but will now fallback to Linux

Reply:
I don't believe they are. Here are the places the symbol is present (`ait` is an alias for `ag --ignore tests`, `ag` is a grep-like):

```
$ ait get_devicelist cloudinit
cloudinit/sources/helpers/digitalocean.py
57:    nics = [f for f in cloudnet.get_devicelist() if cloudnet.is_physical(f)]
```

`cloudnet` here is `cloudinit.net`, so this is calling the `cloudinit.net.get_devicelist` function.

```
cloudinit/net/netplan.py
12:from cloudinit.net import SYS_CLASS_NET, get_devicelist
254:                    for iface in get_devicelist() if
```
The function again.
```
cloudinit/net/__init__.py
309:def get_devicelist():
338:        unstable = [device for device in get_devicelist()
348:    potential_interfaces = set([device for device in get_devicelist()
795:    devs = get_devicelist()
```
All three uses are the function.

```
cloudinit/net/cmdline.py
19:from . import get_devicelist
63:            for k in get_devicelist():
```

`.` is `cloudinit.net` so this is the function.

```
cloudinit/net/dhcp.py
15:    EphemeralIPv4Network, find_fallback_nic, get_devicelist,
127:    elif nic not in get_devicelist():
129:            'Skip dhcp_discovery: nic %s not found in get_devicelist.', nic)
```
Line 15 is an import from `cloudinit.net`, so this is the function.

```
cloudinit/distros/freebsd.py
322:    def get_devicelist():
329:        nics = Distro.get_devicelist()
339:        nics = Distro.get_devicelist()
359:        names = Distro.get_devicelist()
```

These are, of course, the calls that are removed in this PR.

If you look at `cloudinit.distros.Distro`, it doesn't even have a `get_devicelist` method, that's only defined on `cloudinit.distros.freebsd.Distro`. (Is it confusing that all of our `Distro` sub-classes are named `Distro`? Yes, I only just noticed that now!)

Reply:
Yes, the method will never be called on FreeBSD.